src/cloud-api-adaptor/pkg/agent/config.go

```diff
@@ -0,0 +1,50 @@
+package agent
+
+import (
+	"fmt"
+	"strings"
+
+	toml "github.com/pelletier/go-toml/v2"
+)
+
+const (
+	DefaultAaConfigPath = "/run/peerpod/aa.toml"
+)
+
+type TokenConfigs struct {
+	TokenCfg struct {} `toml:"token_configs"`
+	CocoAs struct {
+		URL string `toml:"url"`
+	} `toml:"token_configs.coco_as"`
+	Kbs struct {
+		URL  string `toml:"url"`
+	} `toml:"token_configs.kbs"`
+}
+
+func parseAAKBCParams(aaKBCParams string) (string, error) {
+	parts := strings.SplitN(aaKBCParams, "::", 2)
+	if len(parts) != 2 {
+		return "", fmt.Errorf("Invalid aa-kbs-params input: %s", aaKBCParams)
+	}
+	_, url := parts[0], parts[1]
+	return url, nil
+}
+
+func CreateConfigFile(aaKBCParams string) (string, error) {
+	url, err := parseAAKBCParams(aaKBCParams)
+	if err != nil {
+		return "", err
+	}
+
+	config := TokenConfigs{}
+	config.CocoAs.URL = ""
+	config.Kbs.URL = url
+
+	bytes, err := toml.Marshal(config)
+	if err != nil {
+		return "", err
+	}
+	tomlString := strings.ReplaceAll(string(bytes), "['", "[")
+	tomlString = strings.ReplaceAll(tomlString, "']", "]")
```

Comment:
Why is this required?

Reply:
Because toml marshall generates string like this:
`['token_configs.kbs']`, I dropped the `'` characters in the table.

Reply:
I think it's quoting it because the annotation instructs it to use `.` as a reserved character for nesting.

Why don't we nest the configuration?

```go
type AAConfig struct {
	TokenCfg struct {
		CocoAs struct {
			URL string `toml:"url"`
		} `toml:"coco_as"`
		Kbs struct {
			URL string `toml:"url"`
		} `toml:"kbs"`
	} `toml:"token_configs"`
}
```

this way we don't have to patch the serialized string